File: 10. Queues/MyQueue.js
class CircularQueue {
    constructor(size) {
        this.data = [];
        this.size = size;
        this.length = 0;
        this.front = 0;
        this.rear = -1;
    }
    isEmpty() {
        return (this.length == 0)
    }
    enqueue(element) {
        if (this.length >= this.size)
            console.log("full");
        //rear=(rear+1)%size;  
        this.data[(this.rear + 1) % this.size] = element;// data[0]=10
        this.length++;
    }

    print() {
        for (let i = 0; i < this.data.length; i++) {
            console.log[this.data[i]];
        }
    }

    getfront() {
        if (this.isEmpty()) {
            console.log("no element in circular queue");
        }
        return this.data[(this.front) % (this.size)];
    }

    dequeue() {
        if (this.isEmpty())
            console.log("no element");
        const value = this.getfront();
        this.data[this.front % this.size] = null;
        this.front++;
        this.length--;
        console.log(value);
    }
}

//Driver Code.
let cq = new CircularQueue(5);
cq.enqueue(10);
cq.enqueue(15);
cq.enqueue(16);
cq.enqueue(17);
cq.enqueue(18);
console.log("deleted element ");
cq.dequeue();

File: 10. Queues/myQueue.java
class MyQueue {
    public int front,rear,size;
    public static int arr[];

    MyQueue(int size) {
        this.front = -1;
        this.rear = -1;
        this.size = size;
        arr = new int[size];
    }

    public boolean isEmpty() {
        if (front == -1)
            return true;
        return false;
    }

    public boolean isFull() {
        if (front == 0 && rear == size - 1)
            return true;
        else if (front == rear + 1)
            return true;
        return false;
    }

    public void enQueue(int element) {
        if (isFull()) {
            System.out.println("Queue is full");
        } else {
            if (front == -1)
                front = 0;
            rear = (rear + 1) % size;
            arr[rear] = element;
        }
    }
    
    public void deQueue() {
        int element;
        if (isEmpty()) {
            System.out.println("Queue is empty");
        } else {
            element = arr[front];
            //Condition for single element present in queue.
            if (front == rear) {
                front = -1;
                rear = -1;
            }
            //Checks for more than 1 element in queue. 
            else {
                front = (front + 1) % size;
            }
            System.out.println(element);
        }
    }
}

public class myQueue {

    public static void main(String[] args) {
        MyQueue que = new MyQueue(5);
        que.enQueue(100);
        que.enQueue(200);
        que.enQueue(300);
    
    }
}
